### DISCUSSION TITLE
How to prevent mouse wheel from scrolling

### DISCUSSION THREAD

**Discussion by taylankasap** [i3/i3#6071] [Q&A]
While searching for this, I've noticed that there are only a handful of people who talk about mouse wheel. So let me know if this isn't the responsibility of i3.

What I'm **actually** trying to do is setting the `mod+wheelup -> volume up` and `mod+wheeldown -> volume down`. This part is easy and I already have that. But it still passes the mouse scroll event to the underlying window (which means, mod+wheelup actually does increase the volume **AND** scroll the window). I want to prevent that. How can I do that in i3? And if not, what is the best way to go about it?

These are the bindings I use

```
bindsym --whole-window $mod+button4 exec --no-startup-id pactl set-sink-volume @DEFAULT_SINK@ +2% && $refresh_i3status
bindsym --whole-window $mod+button5 exec --no-startup-id pactl set-sink-volume @DEFAULT_SINK@ -2% && $refresh_i3status
```

**Comment by orestisfl** [comment]
Hmm, I can confirm that which is weird. It's not for all windows because i3 **does** eat the event. For example, uxterm does not scroll for me with your binding. `xev -event mouse` also does not show the scroll events.

However, gnome-terminal, thunar and evince do which means they have a different mechanism to detect scrolls. Binding `button1` works as expected.

**Reply by taylankasap** [reply]
I use wezterm and it doesn't scroll if I use mod+wheel even **without** the binding declared in i3. So maybe uxterm also does not scroll with mod+wheel (without the binding)?

**Reply by orestisfl** [reply]
uxterm does scroll without the binding

**Reply by taylankasap** [reply]
Hmm, then you're right and they have a different detection mechanism. Does that mean this is out of i3's scope?

**Reply by orestisfl** [reply]
Not necessarily, the discussion is especially useful to the i3 community. However, the root cause needs to be found to decide if i3 can handle this case better somehow. I am guessing that this affects other binding daemons/window managers as well though.
